include/mamba/configuration.hpp

```diff
@@ -0,0 +1,1117 @@
+// Copyright (c) 2019, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#ifndef MAMBA_CONFIGURATION_HPP
+#define MAMBA_CONFIGURATION_HPP
+
+#include "mamba/context.hpp"
+#include "mamba/environment.hpp"
+#include "mamba/mamba_fs.hpp"
+#include "mamba/output.hpp"
+#include "mamba/util.hpp"
+
+#include <yaml-cpp/yaml.h>
+
+#include <functional>
+
+
+namespace YAML
+{
+    template <>
+    struct convert<mamba::VerificationLevel>
+    {
+        static Node encode(const mamba::VerificationLevel& rhs)
+        {
+            if (rhs == mamba::VerificationLevel::kDisabled)
+            {
+                return Node("disabled");
+            }
+            else if (rhs == mamba::VerificationLevel::kWarn)
+            {
+                return Node("warn");
+            }
+            else if (rhs == mamba::VerificationLevel::kEnabled)
+            {
+                return Node("enabled");
+            }
+            else
+            {
+                return Node();
+            }
+        }
+
+        static bool decode(const Node& node, mamba::VerificationLevel& rhs)
+        {
+            if (!node.IsScalar())
+            {
+                return false;
+            }
+
+            auto str = node.as<std::string>();
+
+            if (str == "enabled")
+            {
+                rhs = mamba::VerificationLevel::kEnabled;
+            }
+            else if (str == "warn")
+            {
+                rhs = mamba::VerificationLevel::kWarn;
+            }
+            else if (str == "disabled")
+            {
+                rhs = mamba::VerificationLevel::kDisabled;
+            }
+            else
+            {
+                throw std::runtime_error("'VerificationLevel' wrong value");
```

Comment:
I think this should list the accepted values.

Reply:
:+1: